### PR TITLE
Enables end-end timeout with availability strategy on write operations

### PR DESCRIPTION
**Pull request by mbhaskar**
# Description
This PR enables end-end timeout with availability strategy on write operations.
- Create
- Delete
- Upsert
- Patch
- Replace

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by FabianMeiswinkel** [comment]
LGTM - Thanks!

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - Thanks!

**Review by kushagraThapar — CHANGES_REQUESTED** [review]
The below code changes are only for patch API, what about other write operations as mentioned in the PR description. Can you please clarify?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java [hunk 1]
```diff
@@ -2360,8 +2366,6 @@ private Mono<ResourceResponse<Document>> readDocumentInternal(String documentLin
 
             CosmosEndToEndOperationLatencyPolicyConfig endToEndPolicyConfig = getEndToEndOperationLatencyPolicyConfig(options);
 
-            request.requestContext.setEndToEndOperationLatencyPolicyConfig(endToEndPolicyConfig);
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java:2363]: why did we remove this?

Comment (mbhaskar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java:2363]: It is moved to getRxDocumentServiceResponseMonoWithE2ETimeout as it is required for all point operatoins

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java:2363]: Got it, thanks!

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**